crates/subspace-farmer-components/src/plotting.rs

```diff
@@ -315,87 +345,86 @@ where
-                });
-        });
+            })
+            .collect::<Vec<_>>();
+        // Erasure code source record chunks
+        let parity_record_chunks = erasure_coding
+            .extend(&source_record_chunks)
+            .expect("Instance was verified to be able to work with this many values earlier; qed");
+
+        // For every erasure coded chunk check if there is quality present, if so then encode
+        // with PoSpace quality bytes and set corresponding `quality_present` bit to `true`
+        let num_successfully_encoded_chunks = (SBucket::ZERO..=SBucket::MAX)
+            .zip(
+                source_record_chunks
+                    .iter()
+                    .zip(&parity_record_chunks)
+                    .flat_map(|(a, b)| [a, b]),
+            )
+            .zip(encoded_chunks_used.iter_mut())
+            .filter_map(|((s_bucket, record_chunk), mut encoded_chunk_used)| {
+                let quality = pos_table.find_quality(s_bucket.into())?;
+
+                *encoded_chunk_used = true;
+
+                Some(
+                    Simd::from(record_chunk.to_bytes()) ^ Simd::from(quality.create_proof().hash()),
+                )
+            })
+            // Make sure above filter function (and corresponding `encoded_chunk_used` update)
+            // happen at most as many times as there is number of chunks in the record,
+            // otherwise `n+1` iterations could happen and update extra `encoded_chunk_used`
+            // unnecessarily causing issues down the line
+            .take(record.iter().count())
+            .zip(record.iter_mut())
+            // Write encoded chunk back so we can reuse original allocation
+            .map(|(input_chunk, output_chunk)| {
+                *output_chunk = input_chunk.to_array();
+            })
+            .count();
+
+        // In some cases there is not enough PoSpace qualities available, in which case we add
+        // remaining number of unencoded erasure coded record chunks to the end
+        source_record_chunks
+            .iter()
+            .zip(&parity_record_chunks)
+            .flat_map(|(a, b)| [a, b])
+            .zip(encoded_chunks_used.iter())
+            // Skip chunks that were used previously
+            .filter_map(|(record_chunk, encoded_chunk_used)| {
+                if *encoded_chunk_used {
+                    None
+                } else {
+                    Some(record_chunk)
+                }
+            })
+            // First `num_successfully_encoded_chunks` chunks are encoded
+            .zip(record.iter_mut().skip(num_successfully_encoded_chunks))
+            // Write necessary number of unencoded chunks at the end
+            .for_each(|(input_chunk, output_chunk)| {
+                *output_chunk = input_chunk.to_bytes();
+            });
+
+        // Give a chance to interrupt plotting if necessary in between pieces
+        Yield::new().await
```

Comment:
Just a cancellation point is needed. Why not use https://docs.rs/tokio/latest/tokio/task/index.html#yield_now?

Reply:
Hm, I was looking at it and the description seems complicated. But looking at implementation it does what I need, will switch to it instead :+1: